src/main/java/com/google/sps/servlets/SignUpWithGoogleServlet.java

```diff
@@ -0,0 +1,91 @@
+package com.google.sps.servlets;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.Entity;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.rowset.WebRowSet;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.servlet.http.Cookie;
+import java.lang.String;
+import java.nio.charset.*;
+import com.google.cloud.datastore.StructuredQuery.CompositeFilter;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+@WebServlet("/signup")
+public class SignUpWithGoogleServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    String nameValue = Jsoup.clean(request.getParameter("name"), Whitelist.none());
+    String emailValue = Jsoup.clean(request.getParameter("email"), Whitelist.none());
+    String idValue = Jsoup.clean(request.getParameter("ID"),Whitelist.none());
+    System.out.println("hello!"+nameValue);
+
+    Query<Entity> query =
+        Query.newEntityQueryBuilder().setKind("User")
+        .setFilter(
+                PropertyFilter.eq("ID", idValue)
+            )
+        .build();
+    
+    QueryResults<Entity> results = datastore.run(query);
+    System.out.println("result:"+results);
+    if(results.hasNext()){
+        //if the user exits, stop here and set a cookie
```

Comment:
If we really want the name, we might be able to return it to the frontend here. A simple solution: use the result to get the name then add it as a url parameter when we redirect to the main page in the end. Then the frontend grabs the name and displays it.